VDP---Enhanced-Audio-API.md

```diff
@@ -362,6 +378,41 @@ What you hear when this code runs is a note that starts off quiet and over 0.4s
 
+### Type 2: Multi-phase ADSR
+
+`VDU 23, 0, &85, channel, 6, 2, attackCount, [level, duration;]*, sustainCount, [level, duration;]*, releaseCount, [level, duration;]*`
+
+\* this sub-phase data is optional, dependening on the preceding count value, and will repeat as necessary to match the count.
+
+This command will set up a volume envelope that supports multiple sub-phases for each attack/decay, sustain, and release phases.  Conceptually this command is similar to the simpler type 1 ADSR envelope, but it allows each phase to be split into sub-phases for greater control.
+
+This is a variable length command, with the exact format and number of parameters determined by the command itself.
+
+Each sub-phase consists of two pieces of data; a target volume level, and the duration of that sub-phase in the format `level, duration;`.  The duration is a 16-bit value setting the length in ms for that sub-phase.  The target volume level is the envelope data is a number from 0-255 and specifies a relative level centred around 127.  The actual target level, to be reached by the end of that sub-phase, is calculated as follows:
+```
+actualTarget = (noteVolume * targetLevel) / 127
+```
+(This is the same formula that the type 1 ADSR envelope uses for calculating the volume level its the sustain phase.)
+
+Unlike a type 1 ADSR envelope, this envelope type considers "attack" and "decay" to be part of the same envelope phase.  You can simulate distinct "attack" and "decay" phases by providing separate sub-phases for each within the attack/decay phase.
+
+Specifying zero attack/decay sub-phases is equivalent to specifying a single attack sub-phase with target volume level 127 of 0ms duration.
+
+As the "sustain" phase of a volume envelope extends until the note duration has completed, it gets some special consideration.  The following rules will apply:
+
+* if no sustain phases are defined then the sustain level will match the end target level of the attack/decay phase
+* if sustain sub-phases only consist of zero duration entries then the envelope will automatically evenly spread the remaining note duration amongst these sub-phases
+* if any sub-phases contain a duration then the sustain phase will loop through these sub-phases
+    * mixing in zero duration sub-phases will cause abrupt volume level steps
+
+Envelopes with looping sustain phases will always complete a loop cycle before moving on to the release phase.  This means that a note effectively automatically extends its duration.  The audio system will, however, consider the note to be completed once the duration has been reached, even if the note is part-way through a "sustain" loop, allowing playback to be interrupted to start a new note.
+
+Once the sustain phase has completed the envelope will process the release sub-phases.  Conventionally the final release sub-phase will typically target a volume level of 0 for silence, although this is not required.
+
+TODO: add some examples
```

Comment:
I'm going to leave this "TODO" in the docs for now.  I think we're better off merging this documentation update in rather than waiting for more examples to be added